plugins/modules/foreman_hostgroup.py

```diff
@@ -102,14 +131,19 @@
 
 
 # This is the only true source for names (and conversions thereof)
-name_map = {
-    'name': 'name',
-    'description': 'description',
-    'architecture': 'architecture_id',
-    'operatingsystem': 'operatingsystem_id',
-    'media': 'medium_id',
-    'ptable': 'ptable_id',
-    'root_pass': 'root_pass',
+entity_spec = {
+    'id': {},
+    'name': {},
+    'description': {},
+    'architecture': {'type': 'entity', 'flat_name': 'architecture_id'},
+    'architecture_id': {},
```

Comment:
So I dislike this duplication (plus I have the vision that some day we'd use `entity_spec` to generate Ansible's `argument_spec`…).

How about calculating this?

```python
def _flatten_entity(entity, entity_spec):
    """Flatten entity according to spec"""
    extended_entity_spec = {entry['flat_name']:{} for entry in entity_spec.values() if 'flat_name' in entry} # le magic
    extended_entity_spec.update(entity_spec)
    result = {}
    for key, value in entity.items():
        if key in extended_entity_spec:
            spec = extended_entity_spec[key]
            flat_name = spec.get('flat_name', key)
            _type = spec.get('type', 'str')
            if _type == 'entity':
                result[flat_name] = value['id']
            elif _type == 'entity_list':
                result[flat_name] = sorted(val['id'] for val in value)
            else:
                result[flat_name] = value
    return result
```